app_dart/test/model/ci_yaml/target_test.dart

```diff
@@ -240,19 +256,50 @@ void main() {
         expect(target.getDimensions().length, 0);
       });
 
-      test('dimensions exit', () {
-        final Target target = generateTarget(1, properties: <String, String>{'os': 'abc', 'cpu': 'x64'});
+      test('platform dimensions and target dimensions are combined', () {
+        final Target target = generateTarget(
+          1,
+          platform: 'Mac_ios',
+          platformDimensions: <String, String>{
+            'signing_cert': 'none',
+          },
+          properties: <String, String>{'os': 'abc', 'cpu': 'x64'},
+        );
+        final List<RequestedDimension> dimensions = target.getDimensions();
+        expect(dimensions.length, 3);
+        expect(dimensions[0].key, 'signing_cert');
+        expect(dimensions[0].value, 'none');
+        expect(dimensions[1].key, 'os');
+        expect(dimensions[1].value, 'abc');
+        expect(dimensions[2].key, 'cpu');
+        expect(dimensions[2].value, 'x64');
+      });
+
+      test('target specific dimensions overrides platform dimensions', () {
```

Comment:
Can you add a test that the legacy dimensions are still used (such as property cpu), and ensure that the new dimensions fields are prioritized over them